include/method_enums.hpp

```diff
@@ -1,5 +1,14 @@
 #pragma once
 
+//!\brief An enum for the different clustering methods.
+enum detecting_methods
+{
+    cigar_string = 1,
```

Comment:
Why not start with 0?

Reply:
We use 0 for "do not use any method" (see refinement), so I decided to start with 1 here.

Reply:
then you should add `none = 0` :)

Reply:
Sorry, I expressed myself badly. I can't run the algorithm without a detecting method, but without a refinement method. So here 0 simply does not exist but in the refinement there is no_refinement = 0.

Reply:
Well, 0 is the default, so it does exist: https://godbolt.org/z/WGqbMx

IMO, either you enumerate them `0-3` or add a semantic meaning to `0` (`none`).
If you never need to check for `none`, then you can just do `0-3`.

Do you need the numerical values somewhere?
If not, you could omit them altogether.

I would always prefer to use the enum values instead of the numbers, both as user and developer.

Reply:
I understand, then I shift to 0. I need the numbers because I want to allow the user to specify -m 1 or just -m cigar_string.